npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/version_resolver.rb

```diff
@@ -468,13 +481,33 @@ def lockfiles_for_path(lockfiles:, path:)
         def run_checker(path:, version:)
           # If there are both yarn lockfiles and npm lockfiles only run the
           # yarn updater
-          if lockfiles_for_path(lockfiles: dependency_files_builder.yarn_locks, path: path).any?
-            return run_yarn_checker(path: path, version: version)
+          lockfiles = lockfiles_for_path(lockfiles: dependency_files_builder.yarn_locks, path: path)
+          if lockfiles.any?
+            if Helpers.yarn_berry?(lockfiles.first)
+              return run_yarn_berry_checker(path: path, version: version)
+            else
+              return run_yarn_checker(path: path, version: version)
+            end
           end
 
           run_npm_checker(path: path, version: version)
         end
 
+        def run_yarn_berry_checker(path:, version:)
+          SharedHelpers.with_git_configured(credentials: credentials) do
+            Dir.chdir(path) do
+              output = Helpers.run_yarn_command("yarn add #{dependency.name}@#{version}#{Helpers.yarn_berry_args}")
```

Comment:
This is a really clever solution but it took me a while to figure it out. Might be helpful to include a comment explaining how it works